.github/workflows/_charm-tests-integration.yaml

```diff
@@ -28,21 +29,39 @@ jobs:
     with:
       use-charmcraftcache: ${{ inputs.use-charmcraftcache }}
 
+  node-size-validation:
+    runs-on: ubuntu-latest
+    name: Input Node Size validation
+    outputs:
+      size: ${{ steps.node-size.outputs.size}}
+    steps:
+      - name: Validate node size variable
+        if: ${{ !contains(fromJSON('["medium", "large", "xlarge"]'), inputs.node-size) }}
+        run: exit 1
+
   integration-test:
     name: Integration Tests (microk8s)
-    runs-on: ubuntu-22.04
-    needs: build
+    runs-on: [self-hosted, amd64, "${{ inputs.node-size }}", jammy]
+    needs:
+      - build
+      - node-size-validation
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
       - name: Setup operator environment (microk8s)
         uses: charmed-kubernetes/actions-operator@main
         with:
-          juju-channel: 3.4/stable
+          juju-channel: 3.6/stable
           provider: microk8s
-          channel: 1.28-strict/stable
-          bootstrap-options: '--agent-version=3.4.0'
+          channel: 1.32-strict/stable
+          bootstrap-options: '--agent-version=3.6.4'
+          microk8s-addons: "dns hostpath-storage metallb:10.64.140.43-10.64.140.49"
```

Comment:
Only the admin UI and the bundle charms require metallb. Should we make metallb configurable or is it too light so not worth the extra config?

Reply:
i'd say it s not worth the parameterization, let's leave it